addons/website_livechat/models/website_visitor.py

```diff
@@ -107,10 +108,12 @@ def _upsert_visitor(self, access_token, force_track_values=None):
         visitor_id, upsert = super()._upsert_visitor(access_token, force_track_values=force_track_values)
         if upsert == 'inserted':
             visitor_sudo = self.sudo().browse(visitor_id)
-            if discuss_channel_uuid := request.cookies.get("im_livechat_uuid"):
-                discuss_channel = request.env["discuss.channel"].sudo().search([("uuid", "=", discuss_channel_uuid)])
-                discuss_channel.write({
-                    'livechat_visitor_id': visitor_sudo.id,
-                    'anonymous_name': "Visitor #%d (%s)" % (visitor_sudo.id, visitor_sudo.country_id.name) if visitor_sudo.country_id else f"Visitor #{visitor_sudo.id}"
-                })
+            if guest := self.env["mail.guest"]._get_guest_from_context():
+                guest_livechats = guest.channel_ids.filtered(lambda c: c.channel_type == "livechat")
+                guest_livechats.channel_ids.livechat_visitor_id = visitor_sudo.id
+                guest_livechats.channel_ids.anonymous_name = (
```

Comment:
Didn't we reach the point where we can just completely delete this field?
It's just partner name or guest name, and can even be taken from `livechat_visitor_id` on top of it.

Reply:
As it's just a refactoring, I'd like to keep the diff as straightforward as possible. If you don't mind I'd like to do this as a follow up as well.